modules/042-kube-dns/hooks/legacy_metrics.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2021 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hooks
+
+import (
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook/metrics"
+	"github.com/flant/addon-operator/sdk"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	Queue: "metrics",
+	Kubernetes: []go_hook.KubernetesConfig{
+		{
+			Name:       "service",
+			ApiVersion: "v1",
+			Kind:       "Service",
+			FilterFunc: applyLegacyServiceAnnotationFilter,
+		},
+	},
+}, legacyServiceAnnotation)
+
+const (
+	legacyMetricsGroup = "kube_dns_legacy_service_annotation"
+	legacyMetricName   = "d8_kube_dns_deprecated_service_annotation"
+)
+
+func legacyServiceAnnotation(input *go_hook.HookInput) error {
+	input.MetricsCollector.Expire(legacyMetricsGroup)
+
+	snap := input.Snapshots["service"]
+	for _, obj := range snap {
+		if obj == nil {
+			continue
+		}
+
+		svc := obj.(*Service)
+		input.MetricsCollector.Set(legacyMetricName, 1, map[string]string{"namespace": svc.Namespace, "name": svc.Name}, metrics.WithGroup(legacyMetricsGroup))
```

Comment:
Prometheus can override the `namespace` label so that it will become `external_namespace`. Maybe we could specify that this is a service_namespace and service_name?

Reply:
That's an excellent point. Changed it.